contracts/InterchainTokenService.sol

```diff
@@ -756,17 +701,14 @@ contract InterchainTokenService is
             destinationAddress = destinationAddressBytes.toAddress();
         }
 
-        // Return token to the existing express caller
+        address recipient = (expressExecutor == address(0)) ? destinationAddress : expressExecutor;
+        address tokenAddress;
+        (amount, tokenAddress) = _giveToken(tokenId, recipient, amount);
+        // Return token to the express executor
         if (expressExecutor != address(0)) {
-            // slither-disable-next-line unused-return
-            _giveToken(tokenId, expressExecutor, amount);
-
             return;
         }
 
```

Comment:
How about this for better code readability?
```suggestion
        // Return token to the express executor
        if (expressExecutor != address(0)) {
            _giveToken(tokenId, expressExecutor, amount);
            return;
        }

        address tokenAddress;
        (amount, tokenAddress) = _giveToken(tokenId, destinationAddress, amount);
```

Reply:
it's a small optimization. `_giveToken` can be inlined just once, but doesn't seem to have a major impact, so I reverted